apps/auth/oidc/auth.py

```diff
@@ -54,16 +55,21 @@ def authenticate(self, credentials):
             **user,
             'username': username,
             'email': g.oidc_token_info.get('email'),
-            'first_name': g.oidc_token_info.get('given_name'),
-            'last_name': g.oidc_token_info.get('family_name'),
             'display_name': g.oidc_token_info.get('name'),
         }
+
+        # first name and last name is optional in Keycloak
+        if 'given_name' in g.oidc_token_info:
+            sync_data['first_name'] = g.oidc_token_info.get('given_name')
+        if 'family_name' in g.oidc_token_info:
+            sync_data['last_name'] = g.oidc_token_info.get('family_name')
```

Comment:
avoid raising error in https://github.com/superdesk/superdesk-core/blob/3e7ad403ea04b674009dbef04df7dda03f0aae22/superdesk/users/services.py#L111